docs: Teleport Authentication with Microsoft Entra ID (OIDC) (#58715)
gravitational/teleport: docs/pages/zero-trust-access/sso/entra-id-oidc.mdx

Any claim that is present in the OIDC ID Token is preserved as user traits. 
Teleport then starts a new session for the user by issuing a pair of short-lived 
TLS and SSH certificates. These certificates encodes user's identity, 
roles and traits, which are evaluated to grant access in Teleport. 

To demonstrate Microsoft Entra ID group to Teleport role mapping, 
we will have two user groups created in the Microsoft Entra ID tenant: 
1. `ad-app-support`. We will map this group to Teleport preset `requester` role. 
    The preset `requester` role grants permission to request access to resources 
    that are enrolled in Teleport. 
2. `ad-app-admin`. We will map this group to Teleport preset `reviewer` and `editor` 
roles. The preset `editor` role grants permission to manage resource in Teleport. 
The preset `reviewer` grants permission to review access to resources in Teleport.

Instead of creating new groups, you may use an existing Microsoft Entra ID groups 
to follow through this guide. 

## Prerequisites

- Permissions to create an enterprise application, configure and grant Microsoft 
  Graph API permissions and manage groups in the Microsoft Entra ID tenant.

> - Teleport user with a preset `editor` or an equivalent role that allows to 
> read and write Auth Connector, users and roles.

Reviewer: ```suggestion
- Teleport user with a preset `editor` role or an equivalent role that allows you to 
read and write Auth Connectors, users and roles.
```
grammar

- (!docs/pages/includes/tctl.mdx!)

## Step 1/5. Create groups

<Admonition type="info" title="New Group">

>   You may skip this step if you are using an existing Microsoft Entra ID 

Reviewer: ```suggestion
  You may skip this step if you are using existing Microsoft Entra ID 
```
Since "groups" is plural.

>   groups to follow through this guide.

Reviewer: ```suggestion
  groups to follow this guide.
```

in this case), setting up the OIDC claims and a client credential. 

These OIDC configurations are available in the "App registrations" service.

In the Azure Portal, from the “Azure services” menu, select "App registrations".
In the "App registrations" UI, search and select the enterprise application you 
created in Step 2.

### Configure redirect URI

In the "App registrations" UI for the enterprise application, from the “Manage” 
menu, select the “Authentication” menu. In this configuration UI, under the 
"Platform configurations" section, click `+ Add a platform` button. 

OIDC redirect URI for Teleport should point to the OIDC callback endpoint, which 
has the following URI format: `https://example.teleport.sh/v1/webapi/oidc/callback`.

Enter the Teleport Proxy Service hostname of your Teleport cluster below to 
generate the OIDC callback URI. Then copy and paste the value in the Azure Portal
to configure OIDC redirect URI. 
```code
https://<Var name="example.teleport.sh" />/v1/webapi/oidc/callback
```

![Entra ID OIDC redirect URI](../../../img/sso/entraid/oidc/entra-id-oidc-redirect-uri.png)

Save the configuration once the redirect URI is configured.

> ### Setup Client credential 

Reviewer: ```suggestion
### Set up client credentials
```
"setup" is a noun, not a verb, and "client" is a common noun, not a proper noun

Teleport uses OIDC authorization code flow to exchange OIDC authorization code 
for the user's ID Token. To exchange the ID Token, a client credential must be 
set up for Teleport. 

In the "App registrations" UI for the enterprise application, from the "Manage" 
menu, select "Certificates & secrets". Now select the "Client secrets" tab and 
click on the `+ New client secret` button. Create a new client secret. 

![Entra ID OIDC client credential](../../../img/sso/entraid/oidc/entra-id-oidc-client-credential.png)

Once the secret is created, copy the value and save it in your working environment: 

> ```bash 

Reviewer: ```suggestion
```code
```
To use our custom code component
